src/native_main.nim

```diff
@@ -153,21 +171,62 @@ proc handleMessage(msg: MessageRecv): string =
                 reply.code = some(2)
 
         of "move":
-            let src = expandTilde(msg.`from`.get())
-            let dst = expandTilde(msg.to.get())
+            var src = expandTilde(msg.`from`.get())
+            writeLog(">> src == " & $src & "\n")
 
-            if fileExists(dst) or fileExists(joinPath(dst, extractFilename(src))):
-                reply.code = some(1)
-            else:
+            var dst = expandTilde(msg.to.get())
+            writeLog(">> dst == " & $dst & "\n")
+
+            let overwrite = msg.overwrite.get(false)
+            let cleanup = msg.cleanup.get(false)
+
+            if overwrite == false:
+                # Check if 'dst' is a file
+                if fileExists(dst):
+                    writeLog(">> fileExists == " & $dst & "\n")
+                    reply.code = some(1)
+
+                # Check if 'dst' is a directory
+                elif dirExists(dst):
+                    if dst[dst.len - 1] == os.DirSep:
+                        # Remove trailing slash if 'dst' is a directory
+                        let regexStr = os.DirSep & "*$"
+                        dst = dst.replace(rex(regexStr), "")
+                        writeLog(">> dst after regex == " & $dst & "\n")
+
+                    # Prepare final 'dst' with correct path
+                    let srcSplitPath = splitPath(src)
+                    dst = dst & os.DirSep & srcSplitPath.tail
+                    writeLog(">> dst with srcSplitPath == " & $dst & "\n")
+
+                    if fileExists(dst):
+                        writeLog(">> dir+fileExists == " & $dst & "\n")
+                        reply.code = some(1)
```

Comment:
I think this does the same thing as https://github.com/tridactyl/native_messenger/commit/dedffd4f0e3450102457a8628cc108705a27e5ac ?

Reply:
@bovine3dom Looks like it. Let me try and confirm. Thanks.

Reply:
@bovine3dom Squashed in https://github.com/gsbabil/tridactyl_native_messenger/commit/b78790e985541d4749d4ef6fd03e8b4da4500c3d Thanks.